File: tests/files/extract_func_works_after.lua
---@param b string
---@param c string
---@param d table
---@param e table
---@param f table
---@param g function
---@param h string
---@param k string
---@param l string
---@param i string
local function bar(a, b, c, d, e, f, g, h, k, l, i, j)
  a = a + 1
  print(b)
  print(c + 1)
  print(d[1])
  print(e.e)
  print(f:f())
  print(g())
  if h then end
  while k do end
  repeat until l
  i = 'i'
  print(j)

  return a, i
end

---@param a integer
local function foo(a)
  for j = 1, 5 do
    local b = 'b'
    local c = 'c'
    local d = {'d'}
    local e = { e = 'e' }
    local f = {}
    function f.f(self) return 'f' end
    local g, h = function() return 'g' end, 'h'
    local i
    local k = 'k'
    local l = 'l'

    local a, i = bar(a, b, c, d, e, f, g, h, k, l, i, j)

    print(a)
    print(i)
  end
end

File: tests/files/extract_func_works_after.py
class F:
    e = "e"

    def f(self):
        return "f"


def bar(b, c, d, e, f, g, h, k, l, j):
    a = a + 1
    a = a
    a += a
    print(b)
    print(c + "1", sep=c)
    print(d[0])
    print(e["e"])
    print(f.e)
    print(f.f())
    print(g(None))
    g(g)
    if h:
        pass
    while k:
        pass
    for item in l:
      pass
    print(j)
    print(l)

    return a

def foo(a: int, l):
    for j in range(0, 5):
        b = "b"
        c = "c"
        d = ["d"]
        e = {"e": "e"}
        f = F()
        (g, h) = (lambda _g: "g", "h")
        k = "k"

        a = bar(b, c, d, e, f, g, h, k, l, j)

        print(a)
        return j
